File: 50.pow-x-n.cpp
/*
 * @lc app=leetcode id=50 lang=cpp
 *
 * [50] Pow(x, n)
 */

// @lc code=start
#include <stdio.h>      /* printf */
#include <stdlib.h>  
class Solution {
public:
    double myPow(double x, int n) {
        double result = 1;
        if(n == 0) {
            return 1;
        }
        if(x == 1.0) {
            return 1.0;
        }
        if(x == -1.0) {
            if(n&2 == 0) {
                return 1;
            }
            return -1;
        }
        for(int i=0; i<abs(n); i++) {
            result *= x;
        }
        if(n > 0) {
            return result;
        }
        return 1/result;
    }
};
// @lc code=end

File: 50.pow-x-n.py
#
# @lc app=leetcode id=50 lang=python
#
# [50] Pow(x, n)
#

# @lc code=start
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        result = 1
        for _ in range(abs(n)):
            result *= x
        if n >= 0:
            return result
        else:
            return 1/result

# @lc code=end

